core/core-impl/src/main/java/ai/timefold/solver/core/api/solver/SolverManager.java

```diff
@@ -268,11 +364,46 @@ default SolverJob<Solution_, ProblemId_> solveAndListen(ProblemId_ problemId,
      *        If null it defaults to logging the exception as an error.
      * @return never null
      */
+    default SolverJob<Solution_, ProblemId_> solveAndListen(ProblemId_ problemId,
+            Function<? super ProblemId_, ? extends Solution_> problemFinder,
+            Consumer<? super Solution_> bestSolutionConsumer,
+            Consumer<? super Solution_> finalBestSolutionConsumer,
+            BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler) {
+        return builder(problemId, problemFinder, bestSolutionConsumer)
+                .withFinalBestSolutionConsumer(finalBestSolutionConsumer)
+                .withExceptionHandler(exceptionHandler)
+                .solveAndListen();
+    }
+
+    /**
+     * As defined by {@link #solveAndListen(Object, Function, Consumer, Consumer, BiConsumer)}.
+     * <p>
+     * The final best solution is delivered twice:
+     * first to the {@code bestSolutionConsumer} when it is found
+     * and then again to the {@code finalBestSolutionConsumer} when the solver terminates.
+     * Do not store the solution twice.
+     * This allows for use cases that only process the {@link Score} first (during best solution changed events)
+     * and then store the solution upon termination.
+     *
+     * @param problemId never null, an ID for each planning problem. This must be unique.
+     *        Use this problemId to {@link #terminateEarly(Object) terminate} the solver early,
+     *        {@link #getSolverStatus(Object) to get the status} or if the problem changes while solving.
+     * @param problemFinder never null, function that returns a {@link PlanningSolution}, usually with uninitialized planning
+     *        variables
+     * @param bestSolutionConsumer never null, called multiple times, on a consumer thread
+     * @param finalBestSolutionConsumer sometimes null, called only once, at the end, on a consumer thread.
+     *        That final best solution is already consumed by the bestSolutionConsumer earlier.
+     * @param exceptionHandler sometimes null, called if an exception or error occurs.
+     *        If null it defaults to logging the exception as an error.
+     * @param configOverride sometimes null, includes settings that override the default configuration
+     * @return never null
+     */
     SolverJob<Solution_, ProblemId_> solveAndListen(ProblemId_ problemId,
             Function<? super ProblemId_, ? extends Solution_> problemFinder,
             Consumer<? super Solution_> bestSolutionConsumer,
             Consumer<? super Solution_> finalBestSolutionConsumer,
-            BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler);
+            BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler,
+            SolverConfigOverride configOverride);
```

Comment:
The point of the builder is that we would no longer be creating new overloads. This method, therefore, should not exist.

Reply:
Let's wait for the design meeting to define the next steps here.

Reply:
Done!